app/Http/Controllers/Auth/LoginController.php

```diff
@@ -0,0 +1,164 @@
+                            }
+                        } else {
+                            $user_opt = Opt::where('controller_id', $userstatuscheck->id)->where('means', '!=', 'VATUSA API')->where('option', 1)->first();
+                            if ($userstatuscheck->opt != 0 && !isset($user_opt)) {
+                                $opt = new Opt;
+                                $opt->controller_id = $res['cid'];
+                                $opt->ip_address = '0.0.0.0';
+                                $opt->means = 'VATUSA API';
+                                $opt->option = 0;
+                                $opt->save();
+                                $userstatuscheck->opt = 0;
+                            }
+                        }
+                        if ($userstatuscheck->visitor == '1') {
+                            if ($res['facility'] != 'ZZN') {
+                                $userstatuscheck->visitor_from = $res['facility'];
+                            }
+                        } else {
+                            $userstatuscheck->visitor_from = null;
+                        }
+                        $userstatuscheck->save();
+                        $this->completeLogin($resourceOwner, $accessToken);
+                        Auth::loginUsingId($res['cid'], true);
+                    } else {
+                        return redirect('/')->with('error', 'You have not been found on the roster. If you have recently joined, please allow up to an hour for the roster to update.');
+                    }
+                } else {
+                    return redirect('/')->with('error', 'You have not been found on the roster. If you have recently joined, please allow up to an hour for the roster to update.');
+                }
+            } else {
+                return redirect('/')->with('error', 'We are unable to verify your access at this time. Please try again in a few minutes.');
+            }
+            if ($userstatuscheck->status == 0) {
+                return redirect('/dashboard')->with('success', 'You have been logged in successfully. Please note that you are on an LOA and should not control until off the LOA. If this is an error, please let the DATM know.');
+            } else {
+                return redirect()->intended('/dashboard')->with('success', 'You have been logged in successfully.');
+            }
+        } else {
+            return redirect('/')->with('error', 'Bad Signature.');
+        }
+    }
+
+    protected function completeLogin($resourceOwner, $token)
+    {
+        $account = User::firstOrNew(['id' => $resourceOwner->data->cid]);
+        if ($resourceOwner->data->oauth->token_valid === "true") { // User has given us permanent access to data
+            $account->access_token = $token->getToken();
+            $account->refresh_token = $token->getRefreshToken();
+            $account->token_expires = $token->getExpires();
```

Comment:
Why do we need to store this token? If we don't use it, we can probably just not store it for now and add this if we need it in the future.